fix: replace enums with string literal types

bitgoAaron opened the pull request:
- SignatureShareType, CommitmentType, and EncryptedSignerShareType converted to const objects with type exports

Ticket: [CE-5160](https://bitgoinc.atlassian.net/browse/CE-5160)

<!--
# Please be aware of the following when making your pull request:

## Description

Please include a summary of your proposed changes and which issue is being addressed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Issue Number

Internal Users - Please include the related internal tracking number (e.g. BG-000000).
External Users - Please link to any relevant github issues as necessary.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My code compiles correctly for both Node and Browser environments
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My commits follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) and I have properly described any BREAKING CHANGES
- [ ] The ticket or github issue was included in the commit message as a reference
- [ ] I have made corresponding changes to the documentation and on any new/updated functions and/or methods - [jsdoc](https://jsdoc.app/)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
-->


[CE-5160]: https://bitgoinc.atlassian.net/browse/CE-5160?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

kaustubhbitgo commented:
If bundle size, performance is the issue, then instead of introducing a breaking change to replace enums with string literals we can opt for any of the following method.

1. `const enum` - Erases enum type completely and provides only type definition. If used then `tsc` replaces it with string literals.
```typescript
// file.ts
export const enum CommitmentType {
  COMMITMENT = 'commitment',
  DECOMMITMENT = 'decommitment',
}

// example.ts
console.log(CommitmentType.COMMITMENT)

// example.js (after tsc build)
console.log('commitment')
```

2. `as const` - Provides readonly type definitions, less impact on bundle size since it is a direct object mapping instead of function shenanigans that enums do.
```typescript
// file.ts
export const CommitmentType = {
  COMMITMENT: 'commitment',
  DECOMMITMENT: 'decommitment',
} as const;

// example.ts
console.log(CommitmentType.COMMITMENT) /* same in compiled js with less overhead */
```

Have you explored any of these methods?

bitgoAaron committed:
feat: add type exports for base enums

- SignatureShareType, CommitmentType, and
  EncryptedSignerShareType converted to
  const objects with type exports

Ticket: CE-5160

bitgoAaron commented:
> If bundle size, performance is the issue, then instead of introducing a breaking change to replace enums with string literals we can opt for any of the following method.
> 
> 1. `const enum` - Erases enum type completely and provides only type definition. If used then `tsc` replaces it with string literals.
> 
> ```ts
> // file.ts
> export const enum CommitmentType {
>   COMMITMENT = 'commitment',
>   DECOMMITMENT = 'decommitment',
> }
> 
> // example.ts
> console.log(CommitmentType.COMMITMENT)
> 
> // example.js (after tsc build)
> console.log('commitment')
> ```
> 
> 2. `as const` - Provides readonly type definitions, less impact on bundle size since it is a direct object mapping instead of function shenanigans that enums do.
> 
> ```ts
> // file.ts
> export const CommitmentType = {
>   COMMITMENT: 'commitment',
>   DECOMMITMENT: 'decommitment',
> } as const;
> 
> // example.ts
> console.log(CommitmentType.COMMITMENT) /* same in compiled js with less overhead */
> ```
> 
> Have you explored any of these methods?

good call @kaustubhbitgo , went with your suggestion & included a type export which will satisfy our needs tree shaking the UI.
